docs/aws/audit/sgaudit/rules/network_acl_unused_exists.mdx

```diff
@@ -23,6 +23,222 @@ CBP
+4. For each Network ACL, check the "Inbound Rules" and "Outbound Rules" tabs. If there are no rules defined, or if the rules are not associated with any subnets, then the Network ACL is unused and should be removed.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can list all the security groups in your AWS account using the following command:
+
+   ```
+   aws ec2 describe-security-groups --query 'SecurityGroups[*].GroupId' --output text
+   ```
+
+   This command will return a list of all security group IDs.
+
+3. For each security group, you can list the associated network ACLs using the following command:
+
+   ```
+   aws ec2 describe-network-acls --filters Name=group-id,Values=<security-group-id> --query 'NetworkAcls[*].NetworkAclId' --output text
+   ```
+
+   Replace `<security-group-id>` with the ID of the security group. This command will return a list of all network ACL IDs associated with the security group.
+
+4. If a network ACL is not associated with any security group, it is unused and should be removed. You can check this by comparing the list of all network ACLs in your AWS account with the list of network ACLs associated with security groups. If a network ACL is in the first list but not in the second, it is unused.
+
+   To list all network ACLs in your AWS account, use the following command:
+
+   ```
+   aws ec2 describe-network-acls --query 'NetworkAcls[*].NetworkAclId' --output text
+   ```
+
+   This command will return a list of all network ACL IDs. Compare this list with the list of network ACLs associated with security groups to find unused network ACLs.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) in your local environment. Boto3 allows you to directly create, update, and delete AWS services from your Python scripts.
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+3. Create an EC2 resource object using the AWS SDK for Python (Boto3) and retrieve all the security groups.
+
+```python
+ec2_resource = session.resource('ec2')
+
+try:
+    security_groups = ec2_resource.security_groups.all()
+except ClientError as e:
+    print(f"Unexpected error: {e}")
+```
+
+4. Iterate over each security group and check if there are any unused Network ACLs. If there are, print them out.
+
+```python
+for sg in security_groups:
+    if not sg.ip_permissions and not sg.ip_permissions_egress:
+        print(f"Unused Network ACL detected: {sg.group_id}")
+```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>